qoi_image_transport: 1.0.4-1 in 'noetic/distribution.yaml' [bloom]

peci1 opened the pull request:
Increasing version of package(s) in repository `qoi_image_transport` to `1.0.4-1`:

- upstream repository: https://github.com/ctu-vras/qoi_image_transport
- release repository: https://gitlab.fel.cvut.cz/cras/ros-release/qoi_image_transport.git
- distro file: `noetic/distribution.yaml`
- bloom version: `0.12.0`
- previous version for package: `null`

## qoi_image_transport

```
* Fixed licenses.
* CI: Added CI.
* Added unit tests, fixed little bugs.
* Initial commit.
* Contributors: Martin Pecka
```

peci1 committed:
qoi_image_transport: 1.0.1-1 in 'noetic/distribution.yaml' [bloom]

peci1 committed:
Update distribution.yaml

christophebedard commented:
The `package.xml` lists BSD as its license: https://github.com/ctu-vras/qoi_image_transport/blob/3261c512a2a3a54f16ed45b6a2528434f9efe343/package.xml#L12e. It also points to a and points to a `LICENSE` file, which is symlinked to `LICENSES/BSD-3-Clause.txt`. However, there are other licenses in the `LICENSES` directory, notably MIT and Unlicense: https://github.com/ctu-vras/qoi_image_transport/tree/master/LICENSES. I can see that MIT is for at least one header file, but I can't find which files the Unlicense applies to. I think those licenses should also be listed in the `package.xml` file, probably with a comment saying which files they apply to/are for.

christophebedard commented:
- [ ] At least one of the following must be present
  - [ ] Top level license file: TODO clarify
  - [ ] Per package license files:
- [x] License is [OSI-approved](https://opensource.org/licenses): yes (BSD-3, MIT, Unlicense)
- [ ] License correctly listed in package.xmls: TODO clarify
- [x] Public source repo: https://github.com/ctu-vras/qoi_image_transport
- [x] Source repository contains ROS packages
- [ ] Each package meets [REP-144](https://www.ros.org/reps/rep-0144.html) naming conventions

peci1 committed:
Update distribution.yaml

peci1 commented:
Oh, you're right! I somehow always thought there can only be one `<license>` tag... But it makes sense to have more. It's a pity https://github.com/ros-infrastructure/rep/pull/347 hasn't moved too much since it was accepted by the TSC...

The Unlicense was used for test data. However, I reconsidered it and changed to CC0.

The top-level LICENSE symlink is there mostly for Github so that it shows the license in the metadata column.

I've updated this PR with 1.0.3-1 which has the licensing fixed.

christophebedard commented:
Thanks for the update!

As for CC0-1.0, that is tricky. It is not actually a license, but it apparently does have a license-y fallback. However, it is not OSI-approved, which is what we're looking for: https://opensource.org/faq#cc-zero. The thing is that it's only for test data, which is not distributed. I think it might be fine in that case, but let me check with other people.

christophebedard reviewed (changes requested):
Just so that the PR doesn't accidentally get merged before the above is resolved.

peci1 commented:
Does it really make sense to require OSI-approved licenses for data? The software licenses are not designed to be applied to data, altough people often do it. There are other licenses targeted exactly at data, which are e.g. CC licenses, ODbL or PDDL.

christophebedard commented:
It doesn't really make sense, but I think the guideline is "licenses listed in a `package.xml`." Just trying to follow the rules here :grin:

peci1 commented:
Yeah, that's good ;) Please, let me know after you consult this... I can definitely change the license of the test files to BSD, but it would be better to be able to use a more proper license ;)

christophebedard commented:
Of course, I'll let you know. I've added this to the list of rosdistro PRs to discuss for next week's PMC meeting, but I may get  a second opinion before then (hopefully).

christophebedard commented:
We just talked about this in today's ROS PMC meeting (2025-04-01).

You can simply drop CC0 from the `package.xml`, since the test data is not distributed, but still keep it somewhere in the repo, which you already have as `.bag.license` files next to the test bag files.

peci1 committed:
Update distribution.yaml

christophebedard reviewed (approved):
Thanks for making the changes!

ros-discourse commented:
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/preparing-for-noetic-sync-2025-04-03/42940/2
